crates/resolver/src/name_server/name_server_pool.rs

```diff
@@ -297,8 +298,23 @@ where
         // construct the parallel requests, 2 is the default
         let mut par_conns = SmallVec::<[NameServer<C, P>; 2]>::new();
         let count = conns.len().min(opts.num_concurrent_reqs.max(1));
-        for conn in conns.drain(..count) {
-            par_conns.push(conn);
+
+        if opts.shuffle_dns_servers {
+            for _ in 0..count {
+                if conns.is_empty() {
```

Comment:
I think the invariant is forced later in the code. But it can probably be extracted from the multiple places that need it to once at the top of the function.

Reply:
I checked this and apparently, it isn't enforced anywhere. I still doubt that it is not checked, my guess is that some struct items need to have a value and `conns` will never be empty at this point so the `.is_empty()` check is redundant.